From 35bc7f2574d76277b9725870f529a70709f5984b Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Wed, 30 Nov 2022 08:32:11 +0200
Subject: [PATCH] [SW-112227] erom: fail secondary image load

In erom, while loading ppboot secondary image need to
check SEC_DIS efuse bit. If SEC_DIS bit is set, shouldn't
load secondary image

Change-Id: I7dd4290c7e24de8329302386e508dd90d92bde14
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/app/ext_rom/erom_pcie.c b/app/ext_rom/erom_pcie.c
index a47c627..22ef76a 100644
--- a/app/ext_rom/erom_pcie.c
+++ b/app/ext_rom/erom_pcie.c
@@ -138,6 +138,13 @@
 	uint32_t flash_image_offset;
 	uint32_t flash_ppboot_pcie_offset;
 	uint32_t ppboot_pcie_size;
+	uint32_t config_fuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
+
+	if ((image_ptr == ARCBOOT_SECONDARY) &&
+		(config_fuse & EFUSE_SYSCFG_DISABLE_SECONDARY_IMG)) {
+		preboot_puts("preboot secondary image disabled!\n");
+		return ARCBOOT_BRINGUP_ERROR;
+	}
 
 	flash_image_offset = arcboot_flash_image_offset(image_ptr);
 	/* check if offset is within safe limits */
